--- a/src/mapi/Makefile.am	2016-04-17 13:16:14.000000000 -0500
+++ b/src/mapi/Makefile.am	2016-05-05 14:41:24.823647000 -0500
@@ -116,6 +116,9 @@ glapi_libglapi_la_CPPFLAGS = \
 if HAVE_APPLEDRI
 glapi_libglapi_la_SOURCES += glapi/glapi_gentable.c
 endif
+if HAVE_WINDOWSDRI
+glapi_libglapi_la_SOURCES += glapi/glapi_gentable.c
+endif
 
 if HAVE_SHARED_GLAPI
 glapi_libglapi_la_SOURCES += $(MAPI_BRIDGE_FILES) glapi/glapi_mapi_tmp.h
--- a/src/mapi/glapi/glapi.h	2016-04-17 13:16:14.000000000 -0500
+++ b/src/mapi/glapi/glapi.h	2016-05-05 14:30:09.439384100 -0500
@@ -158,7 +158,7 @@ _GLAPI_EXPORT const char *
 _glapi_get_proc_name(unsigned int offset);
 
 
-#ifdef GLX_USE_APPLEGL
+#if defined(GLX_USE_APPLEGL) || defined(GLX_USE_WINDOWSGL)
 _GLAPI_EXPORT struct _glapi_table *
 _glapi_create_table_from_handle(void *handle, const char *symbol_prefix);
 #endif
