--- origsrc/Mesa-10.2.3/src/gallium/auxiliary/os/os_process.c	2014-01-31 14:40:50.000000000 -0600
+++ src/Mesa-10.2.3/src/gallium/auxiliary/os/os_process.c	2014-07-20 11:05:11.628341500 -0500
@@ -32,7 +32,7 @@
 
 #if defined(PIPE_SUBSYSTEM_WINDOWS_USER)
 #  include <windows.h>
-#elif defined(__GLIBC__)
+#elif defined(__GLIBC__) || defined(__CYGWIN__)
 #  include <errno.h>
 #elif defined(PIPE_OS_BSD) || defined(PIPE_OS_APPLE)
 #  include <stdlib.h>
@@ -68,7 +68,7 @@ os_get_process_name(char *procname, size
 
    name = lpProcessName;
 
-#elif defined(__GLIBC__)
+#elif defined(__GLIBC__) || defined(__CYGWIN__)
    name = program_invocation_short_name;
 #elif defined(PIPE_OS_BSD) || defined(PIPE_OS_APPLE)
    /* *BSD and OS X */
