--- origsrc/Mesa-10.3.3/src/gallium/targets/dri/Makefile.am	2014-11-08 10:03:25.000000000 -0600
+++ src/Mesa-10.3.3/src/gallium/targets/dri/Makefile.am	2014-11-16 12:33:23.316914300 -0600
@@ -26,6 +26,7 @@ gallium_dri_la_LDFLAGS = \
 	-shrext .so \
 	-module \
 	-avoid-version \
+	-no-undefined \
 	$(GC_SECTIONS)
 
 if HAVE_LD_VERSION_SCRIPT
@@ -190,6 +191,12 @@ gallium_dri_la_LIBADD += \
 
 endif # HAVE_GALLIUM_STATIC_TARGETS
 
+if HAVE_SHARED_GLAPI
+gallium_dri_la_LIBADD += $(top_builddir)/src/mapi/shared-glapi/libglapi.la
+else
+gallium_dri_la_LIBADD += $(top_builddir)/src/mapi/glapi/libglapi.la
+endif
+
 if HAVE_MESA_LLVM
 gallium_dri_la_LIBADD += $(LLVM_LIBS)
 gallium_dri_la_LDFLAGS += $(LLVM_LDFLAGS)
--- origsrc/Mesa-10.3.3/src/mesa/drivers/dri/Makefile.am	2014-11-04 18:57:33.000000000 -0600
+++ src/Mesa-10.3.3/src/mesa/drivers/dri/Makefile.am	2014-11-16 12:34:16.766988100 -0600
@@ -51,7 +51,7 @@ driinclude_HEADERS = $(top_srcdir)/inclu
 nodist_EXTRA_mesa_dri_drivers_la_SOURCES = dummy.cpp
 mesa_dri_drivers_la_SOURCES =
 mesa_dri_drivers_la_LDFLAGS = \
-        -module -avoid-version -shared -shrext .so \
+        -module -avoid-version -shared -shrext .so -no-undefined \
         $(BSYMBOLIC) \
         $(GC_SECTIONS) \
         $()
@@ -63,6 +63,12 @@ mesa_dri_drivers_la_LIBADD = \
         $(DRI_LIB_DEPS) \
         $()
 
+if HAVE_SHARED_GLAPI
+mesa_dri_drivers_la_LIBADD += $(top_builddir)/src/mapi/shared-glapi/libglapi.la
+else
+mesa_dri_drivers_la_LIBADD += $(top_builddir)/src/mapi/glapi/libglapi.la
+endif
+
 if NEED_MEGADRIVER
 dri_LTLIBRARIES = mesa_dri_drivers.la
 
