--- origsrc/mesa-10.5.4/src/egl/drivers/dri2/egl_dri2.c	2015-04-24 16:09:35.000000000 -0500
+++ src/mesa-10.5.4/src/egl/drivers/dri2/egl_dri2.c	2015-05-01 03:05:53.425694600 -0500
@@ -2132,6 +2132,8 @@ dri2_load(_EGLDriver *drv)
 #ifdef HAVE_SHARED_GLAPI
 #ifdef HAVE_ANDROID_PLATFORM
    const char *libname = "libglapi.so";
+#elif defined(__CYGWIN__)
+   const char *libname = "cygglapi-0.dll";
 #else
    const char *libname = "libglapi.so.0";
 #endif
--- origsrc/mesa-10.5.4/src/gallium/auxiliary/util/u_format_s3tc.c	2014-10-02 22:59:50.000000000 -0500
+++ src/mesa-10.5.4/src/gallium/auxiliary/util/u_format_s3tc.c	2015-04-30 21:08:45.106143600 -0500
@@ -32,6 +32,8 @@
 
 #if defined(_WIN32) || defined(WIN32)
 #define DXTN_LIBNAME "dxtn.dll"
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #elif defined(__APPLE__)
 #define DXTN_LIBNAME "libtxc_dxtn.dylib"
 #else
--- origsrc/mesa-10.5.4/src/glx/dri_common.c	2015-04-15 04:41:52.000000000 -0500
+++ src/mesa-10.5.4/src/glx/dri_common.c	2015-04-30 21:09:28.559661500 -0500
@@ -98,8 +98,14 @@ driOpenDriver(const char *driverName)
    char realDriverName[200];
    int len;
 
+#ifdef __CYGWIN__
+#define LIBGL_SO "cygGL-1.dll"
+#else
+#define LIBGL_SO "libGL.so.1"
+#endif
+
    /* Attempt to make sure libGL symbols will be visible to the driver */
-   glhandle = dlopen("libGL.so.1", RTLD_NOW | RTLD_GLOBAL);
+   glhandle = dlopen(LIBGL_SO, RTLD_NOW | RTLD_GLOBAL);
 
    libPaths = NULL;
    if (geteuid() == getuid()) {
--- origsrc/mesa-10.5.4/src/mesa/main/texcompress_s3tc.c	2015-04-24 16:09:36.000000000 -0500
+++ src/mesa-10.5.4/src/mesa/main/texcompress_s3tc.c	2015-04-30 21:09:50.226912900 -0500
@@ -51,6 +51,8 @@
 #define DXTN_LIBNAME "dxtn.dll"
 #define RTLD_LAZY 0
 #define RTLD_GLOBAL 0
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #else
 #define DXTN_LIBNAME "libtxc_dxtn.so"
 #endif
