--- origsrc/mesa-10.6.9/src/egl/drivers/dri2/egl_dri2.c	2015-10-03 06:30:41.000000000 -0500
+++ src/mesa-10.6.9/src/egl/drivers/dri2/egl_dri2.c	2015-10-07 17:21:11.240896000 -0500
@@ -2338,6 +2338,8 @@ dri2_load(_EGLDriver *drv)
 #ifdef HAVE_SHARED_GLAPI
 #ifdef HAVE_ANDROID_PLATFORM
    const char *libname = "libglapi.so";
+#elif defined(__CYGWIN__)
+   const char *libname = "cygglapi-0.dll";
 #else
    const char *libname = "libglapi.so.0";
 #endif
--- origsrc/mesa-10.6.9/src/gallium/auxiliary/util/u_format_s3tc.c	2015-10-03 06:30:41.000000000 -0500
+++ src/mesa-10.6.9/src/gallium/auxiliary/util/u_format_s3tc.c	2015-10-07 17:21:11.247896400 -0500
@@ -32,6 +32,8 @@
 
 #if defined(_WIN32) || defined(WIN32)
 #define DXTN_LIBNAME "dxtn.dll"
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #elif defined(__APPLE__)
 #define DXTN_LIBNAME "libtxc_dxtn.dylib"
 #else
--- origsrc/mesa-10.6.9/src/glx/dri_common.c	2015-10-07 17:19:20.838581400 -0500
+++ src/mesa-10.6.9/src/glx/dri_common.c	2015-10-07 17:20:46.890503300 -0500
@@ -73,6 +73,9 @@ dri_message(int level, const char *f, ..
    }
 }
 
+#ifdef __CYGWIN__
+#define GL_LIB_NAME "cygGL-1.dll"
+#endif
 #ifndef GL_LIB_NAME
 #define GL_LIB_NAME "libGL.so.1"
 #endif
--- origsrc/mesa-10.6.9/src/mesa/main/texcompress_s3tc.c	2015-10-02 03:45:52.000000000 -0500
+++ src/mesa-10.6.9/src/mesa/main/texcompress_s3tc.c	2015-10-07 17:21:11.252896700 -0500
@@ -46,6 +46,8 @@
 #define DXTN_LIBNAME "dxtn.dll"
 #define RTLD_LAZY 0
 #define RTLD_GLOBAL 0
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #else
 #define DXTN_LIBNAME "libtxc_dxtn.so"
 #endif
