--- origsrc/mesa-11.0.2/src/egl/drivers/dri2/egl_dri2.c	2015-09-28 14:23:46.000000000 -0500
+++ src/mesa-11.0.2/src/egl/drivers/dri2/egl_dri2.c	2015-10-08 00:24:49.897985900 -0500
@@ -2432,6 +2432,8 @@ dri2_load(_EGLDriver *drv)
    const char *libname = "libglapi.so";
 #elif defined(__APPLE__)
    const char *libname = "libglapi.0.dylib";
+#elif defined(__CYGWIN__)
+   const char *libname = "cygglapi-0.dll";
 #else
    const char *libname = "libglapi.so.0";
 #endif
--- origsrc/mesa-11.0.2/src/gallium/auxiliary/util/u_format_s3tc.c	2015-09-25 10:13:06.000000000 -0500
+++ src/mesa-11.0.2/src/gallium/auxiliary/util/u_format_s3tc.c	2015-10-08 00:27:27.994641800 -0500
@@ -32,6 +32,8 @@
 
 #if defined(_WIN32) || defined(WIN32)
 #define DXTN_LIBNAME "dxtn.dll"
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #elif defined(__APPLE__)
 #define DXTN_LIBNAME "libtxc_dxtn.dylib"
 #else
--- origsrc/mesa-11.0.3/src/glx/dri_common.c	2015-10-14 01:01:46.708782700 -0500
+++ src/mesa-11.0.3/src/glx/dri_common.c	2015-10-14 01:06:56.378605800 -0500
@@ -73,6 +73,9 @@ dri_message(int level, const char *f, ..
    }
 }
 
+#ifdef __CYGWIN__
+#define GL_LIB_NAME "cygGL-1.dll"
+#endif
 #ifndef GL_LIB_NAME
 #define GL_LIB_NAME "libGL.so.1"
 #endif
--- origsrc/mesa-11.0.2/src/mesa/main/texcompress_s3tc.c	2015-09-23 15:04:15.000000000 -0500
+++ src/mesa-11.0.2/src/mesa/main/texcompress_s3tc.c	2015-10-08 00:27:28.004641800 -0500
@@ -46,6 +46,8 @@
 #define DXTN_LIBNAME "dxtn.dll"
 #define RTLD_LAZY 0
 #define RTLD_GLOBAL 0
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #else
 #define DXTN_LIBNAME "libtxc_dxtn.so"
 #endif
