--- origsrc/mesa-11.1.2/configure.ac	2016-04-12 18:38:35.141083800 -0500
+++ src/mesa-11.1.2/configure.ac	2016-04-12 18:39:45.507519200 -0500
@@ -241,15 +241,12 @@ dnl Compiler macros
 DEFINES="-D__STDC_LIMIT_MACROS"
 AC_SUBST([DEFINES])
 case "$host_os" in
-linux*|*-gnu*|gnu*)
+linux*|*-gnu*|gnu*|cygwin*)
     DEFINES="$DEFINES -D_GNU_SOURCE"
     ;;
 solaris*)
     DEFINES="$DEFINES -DSVR4"
     ;;
-cygwin*)
-    DEFINES="$DEFINES -D_XOPEN_SOURCE=700"
-    ;;
 esac
 
 dnl Add flags for gcc and g++
