--- origsrc/mesa-13.0.3/src/gallium/drivers/swr/Makefile.am	2017-01-23 23:39:44.805368400 -0600
+++ src/mesa-13.0.3/src/gallium/drivers/swr/Makefile.am	2017-01-23 21:57:42.699756900 -0600
@@ -146,14 +146,20 @@ rasterizer/archrast/gen_ar_eventhandlerf
 		--output rasterizer/archrast/gen_ar_eventhandlerfile.h \
 		--gen_eventhandlerfile_h
 
+if HAVE_SHARED_GLAPI
+SHARED_GLAPI_LIB = $(top_builddir)/src/mapi/shared-glapi/libglapi.la
+endif
+
 COMMON_LIBADD = \
 	$(top_builddir)/src/gallium/auxiliary/libgallium.la \
 	$(top_builddir)/src/mesa/libmesagallium.la \
+	$(SHARED_GLAPI_LIB) \
 	$(LLVM_LIBS)
 
 COMMON_LDFLAGS = \
 	-shared \
 	-module \
+	-avoid-version \
 	-no-undefined \
 	$(GC_SECTIONS) \
 	$(NO_UNDEFINED) \
--- origsrc/mesa-13.0.3/src/gallium/targets/osmesa/Makefile.am	2016-11-10 16:05:17.000000000 -0600
+++ src/mesa-13.0.3/src/gallium/targets/osmesa/Makefile.am	2017-01-23 23:24:38.401629800 -0600
@@ -63,7 +63,6 @@ lib@OSMESA_LIB@_la_LIBADD = \
 	$(top_builddir)/src/gallium/drivers/trace/libtrace.la \
 	$(top_builddir)/src/gallium/drivers/softpipe/libsoftpipe.la \
 	$(top_builddir)/src/gallium/state_trackers/osmesa/libosmesa.la \
-	$(top_builddir)/src/mapi/glapi/libglapi.la \
 	$(SHARED_GLAPI_LIB) \
 	$(OSMESA_LIB_DEPS) \
 	$(CLOCK_LIB)
