--- origsrc/mesa-18.0.2/src/gallium/targets/osmesa/meson.build	2018-04-28 11:25:16.000000000 -0500
+++ src/mesa-18.0.2/src/gallium/targets/osmesa/meson.build	2018-04-29 17:06:05.446545800 -0500
@@ -43,9 +43,9 @@ libosmesa = shared_library(
     inc_gallium_drivers,
   ],
   link_depends : osmesa_link_deps,
-  link_whole : [libosmesa_st],
+  link_whole : [libglapi_static, libosmesa_st],
   link_with : [
-    libmesa_gallium, libgallium, libglapi_static, libws_null, libtrace,
+    libmesa_gallium, libgallium, libws_null, libtrace,
     osmesa_link_with,
   ],
   dependencies : [
