--- origsrc/Mesa-7.11.2/src/gallium/state_trackers/vega/api_filters.c	2011-10-14 19:43:58.000000000 -0500
+++ src/Mesa-7.11.2/src/gallium/state_trackers/vega/api_filters.c	2011-12-25 13:44:47.373646500 -0600
@@ -24,6 +24,7 @@
  *
  **************************************************************************/
 
+#undef __STRICT_ANSI__  /* for M_PI */
 #include "VG/openvg.h"
 
 #include "vg_context.h"
--- origsrc/Mesa-7.11.2/src/gallium/state_trackers/vega/bezier.c	2010-03-30 17:40:35.000000000 -0500
+++ src/Mesa-7.11.2/src/gallium/state_trackers/vega/bezier.c	2011-12-25 13:45:19.150901900 -0600
@@ -24,6 +24,7 @@
  *
  **************************************************************************/
 
+#undef __STRICT_ANSI__  /* for M_PI */
 #include "bezier.h"
 
 #include "matrix.h"
