--- origsrc/Mesa-7.2/include/GL/internal/dri_interface.h	2008-08-25 09:46:41.000000000 -0500
+++ src/Mesa-7.2/include/GL/internal/dri_interface.h	2008-10-29 00:18:29.107887900 -0500
@@ -41,7 +41,7 @@
 #define DRI_INTERFACE_H
 
 /* Make this something other than __APPLE__ for other arcs with no drm.h */
-#ifndef __APPLE__
+#if !defined(__APPLE__) && !defined(__CYGWIN__)
 #include <drm.h>
 #else
 typedef unsigned int drm_context_t;
