--- origsrc/Mesa-7.2/bin/mklib	2008-10-26 18:15:00.956410900 -0500
+++ src/Mesa-7.2/bin/mklib	2008-10-26 16:57:50.000000000 -0500
@@ -833,11 +833,11 @@
             # finish up
             FINAL_LIBS=${LIBNAME}.a
         else
-	    OPTS="-shared -Wl,-export-all -Wl,--out-implib=${LIBNAME}-${MAJOR}.dll.a"
+	    OPTS="-shared -Wl,--enable-auto-image-base -Wl,-export-all -Wl,--out-implib=${LIBNAME}.dll.a"
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
             fi
-            echo "mklib: Making" $ARCH "shared library: " ${LIBNAME}-${MAJOR}.dll
+            echo "mklib: Making" $ARCH "shared library: " ${CYGNAME}-${MAJOR}.dll
 
             if [ $CPLUSPLUS = 1 ] ; then
                 LINK="g++"
@@ -846,16 +846,14 @@
             fi
 
             # rm any old libs
-            rm -f ${LIBNAME}-${MAJOR}.dll
+            rm -f ${CYGNAME}-${MAJOR}.dll
             rm -f ${LIBNAME}.dll.a
             rm -f ${LIBNAME}.a
 
             # make lib
             ${LINK} ${OPTS} ${LDFLAGS} -o ${CYGNAME}-${MAJOR}.dll ${OBJECTS} ${DEPS}
-            # make usual symlinks
-            ln -s ${LIBNAME}-${MAJOR}.dll.a ${LIBNAME}.dll.a
             # finish up
-            FINAL_LIBS="${LIBNAME}-${MAJOR}.dll.a ${LIBNAME}.dll.a"
+            FINAL_LIBS="${LIBNAME}.dll.a"
 	    # special case for installing in bin
             FINAL_BINS="${CYGNAME}-${MAJOR}.dll"
         fi
--- origsrc/Mesa-7.2/progs/xdemos/ipc.c	2008-08-25 09:46:42.000000000 -0500
+++ src/Mesa-7.2/progs/xdemos/ipc.c	2008-10-26 17:08:44.706145900 -0500
@@ -29,7 +29,6 @@
 #include <string.h>
 #include <sys/types.h>
 #include <netinet/in.h>
-#include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <unistd.h>
