--- origsrc/Mesa-7.5.2/include/EGL/eglplatform.h	2009-09-25 18:54:04.000000000 -0500
+++ src/Mesa-7.5.2/include/EGL/eglplatform.h	2009-10-13 21:00:18.169548900 -0500
@@ -69,7 +69,7 @@ typedef HWND	NativeWindowType;
 typedef HBITMAP NativePixmapType;
 /** END Added for Windows **/
 
-#elif defined(__gnu_linux__) || defined(__FreeBSD__) || defined(__sun) || defined(__APPLE__)
+#elif defined(__gnu_linux__) || defined(__FreeBSD__) || defined(__sun) || defined(__APPLE__) || defined(__CYGWIN__)
 
 
 /** BEGIN Added for X (Mesa) **/
--- origsrc/Mesa-7.5.2/src/egl/drivers/demo/Makefile	2009-08-23 22:52:57.000000000 -0500
+++ src/Mesa-7.5.2/src/egl/drivers/demo/Makefile	2009-10-13 21:15:00.352087200 -0500
@@ -23,7 +23,7 @@ default: $(TOP)/$(LIB_DIR)/demodriver.so
 $(TOP)/$(LIB_DIR)/demodriver.so: $(OBJECTS)
 	$(MKLIB) -o demodriver.so -noprefix -linker '$(CC)' \
 		-ldflags '$(LDFLAGS)' -install $(TOP)/$(LIB_DIR) \
-		$(OBJECTS)
+		$(OBJECTS) -L$(TOP)/$(LIB_DIR) -lEGL
 
 install:
 
