--- origsrc/Mesa-7.8.2/progs/egl/Makefile	2010-06-15 12:43:41.000000000 -0500
+++ src/Mesa-7.8.2/progs/egl/Makefile	2010-10-04 13:51:43.930848800 -0500
@@ -7,7 +7,6 @@ include $(TOP)/configs/current
 INCLUDE_DIRS = -I$(TOP)/include $(X11_CFLAGS)
 
 HEADERS = $(TOP)/include/GLES/egl.h
-LIB_DEP = $(TOP)/$(LIB_DIR)/libEGL.so
 
 LIBS = -L$(TOP)/$(LIB_DIR) -lEGL -lGL
 
--- origsrc/Mesa-7.8.2/src/egl/drivers/glx/Makefile	2010-04-27 16:41:59.000000000 -0500
+++ src/Mesa-7.8.2/src/egl/drivers/glx/Makefile	2010-10-04 13:48:59.509066600 -0500
@@ -11,6 +11,6 @@ EGL_INCLUDES = \
 	-I$(TOP)/src/egl/main
 
 EGL_CFLAGS = $(X11_CFLAGS)
-EGL_LIBS = $(X11_LIBS) -lGL
+EGL_LIBS = $(X11_LIBS) -lEGL -lGL
 
 include ../Makefile.template
