--- origsrc/Mesa-7.8.2/src/mesa/Makefile	2010-06-15 12:43:42.000000000 -0500
+++ src/Mesa-7.8.2/src/mesa/Makefile	2010-10-04 14:32:18.666261200 -0500
@@ -43,6 +43,7 @@ libglapi.a: $(GLAPI_OBJECTS)
 ######################################################################
 # Device drivers
 driver_subdirs: libmesa.a libglapi.a libmesagallium.a
+	@ (cd ../glx && $(MAKE))
 	@ (cd drivers && $(MAKE))
 
 
--- origsrc/Mesa-7.8.2/src/mesa/drivers/dri/Makefile.template	2010-06-15 12:43:42.000000000 -0500
+++ src/Mesa-7.8.2/src/mesa/drivers/dri/Makefile.template	2010-10-04 13:47:28.306655200 -0500
@@ -1,6 +1,6 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
+MESA_MODULES = $(TOP)/src/mesa/libmesa.a $(TOP)/lib/libGL.dll.a
 
 COMMON_GALLIUM_SOURCES = \
         ../common/utils.c \
