--- origsrc/Mesa-8.0.1/src/egl/drivers/glx/egl_glx.c
+++ src/Mesa-8.0.1/src/egl/drivers/glx/egl_glx.c
@@ -1076,7 +1076,11 @@ GLX_Load(_EGLDriver *drv)
    if (!GLX_drv->glXGetProcAddress)
       GLX_drv->glXGetProcAddress = dlsym(RTLD_DEFAULT, "glXGetProcAddressARB");
    if (!GLX_drv->glXGetProcAddress) {
+#ifdef __CYGWIN__
+      handle = dlopen("cygGL-1.dll", RTLD_LAZY | RTLD_LOCAL);
+#else
       handle = dlopen("libGL.so", RTLD_LAZY | RTLD_LOCAL);
+#endif
       if (!handle)
          goto fail;
 
--- origsrc/Mesa-7.11.2/src/glx/dri_common.c	2011-12-23 02:05:00.515288100 -0600
+++ src/Mesa-7.11.2/src/glx/dri_common.c	2011-12-23 00:39:51.000000000 -0600
@@ -104,8 +104,14 @@ driOpenDriver(const char *driverName)
    char realDriverName[200];
    int len;
 
+#ifdef __CYGWIN__
+#define LIBGL_SO "cygGL-1.dll"
+#else
+#define LIBGL_SO "libGL.so.1"
+#endif
+
    /* Attempt to make sure libGL symbols will be visible to the driver */
-   glhandle = dlopen("libGL.so.1", RTLD_NOW | RTLD_GLOBAL);
+   glhandle = dlopen(LIBGL_SO, RTLD_NOW | RTLD_GLOBAL);
 
    libPaths = NULL;
    if (geteuid() == getuid()) {
--- origsrc/Mesa-7.11.2/src/mesa/main/texcompress_s3tc.c	2011-12-23 02:05:00.518288200 -0600
+++ src/Mesa-7.11.2/src/mesa/main/texcompress_s3tc.c	2011-12-23 02:17:11.418093300 -0600
@@ -55,6 +55,8 @@
 #define RTLD_GLOBAL 0
 #elif defined(__DJGPP__)
 #define DXTN_LIBNAME "dxtn.dxe"
+#elif defined(__CYGWIN__)
+#define DXTN_LIBNAME "cygtxc_dxtn.dll"
 #else
 #define DXTN_LIBNAME "libtxc_dxtn.so"
 #endif
