--- origsrc/Mesa-7.11.2/src/gallium/targets/Makefile.dri	2011-10-14 19:43:58.000000000 -0500
+++ src/Mesa-7.11.2/src/gallium/targets/Makefile.dri	2011-12-25 13:51:09.605513000 -0600
@@ -34,6 +34,9 @@ OBJECTS = \
 	$(C_SOURCES:.c=.o) \
 	$(ASM_SOURCES:.S=.o) 
 
+ifeq ($(SHARED_GLAPI),1)
+DRI_LIB_DEPS += -L$(TOP)/$(LIB_DIR) -l$(GLAPI_LIB)
+endif
 
 ### Include directories
 SHARED_INCLUDES = \
--- origsrc/Mesa-8.0.2/src/mapi/glapi/gen/Makefile	2012-03-21 10:54:27.000000000 -0500
+++ src/Mesa-8.0.2/src/mapi/glapi/gen/Makefile	2012-04-03 20:45:30.492467000 -0500
@@ -150,7 +150,7 @@ $(XORG_GLAPI_DIR)/%.h: $(MESA_GLAPI_DIR)
 
 $(MESA_GLAPI_DIR)/glapi_mapi_tmp.h: $(MESA_MAPI_DIR)/mapi_abi.py $(COMMON_ES)
 	$(PYTHON2) $(PYTHON_FLAGS) $< \
-		--printer glapi --mode lib gl_and_es_API.xml > $@
+		--printer glapi --mode lib gl_API.xml > $@
 
 $(MESA_GLAPI_DIR)/glprocs.h: gl_procs.py $(COMMON)
 	$(PYTHON2) $(PYTHON_FLAGS) $< > $@
--- origsrc/Mesa-8.0.2/src/mapi/shared-glapi/Makefile	2012-03-21 10:54:27.000000000 -0500
+++ src/Mesa-8.0.2/src/mapi/shared-glapi/Makefile	2012-04-03 21:31:25.291032400 -0500
@@ -35,7 +35,7 @@ $(glapi_OBJECTS): %.o: $(MAPI)/%.c
 $(glapi_SOURCES): glapi_mapi_tmp.h
 
 include $(GLAPI)/gen/glapi_gen.mk
-glapi_mapi_tmp.h: $(GLAPI)/gen/gl_and_es_API.xml $(glapi_gen_mapi_deps)
+glapi_mapi_tmp.h: $(GLAPI)/gen/gl_API.xml $(glapi_gen_mapi_deps)
 	$(call glapi_gen_mapi,$<,shared-glapi)
 
 .PHONY: clean
--- origsrc/Mesa-7.11.2/src/mesa/drivers/dri/Makefile.defines	2011-10-14 19:43:58.000000000 -0500
+++ src/Mesa-7.11.2/src/mesa/drivers/dri/Makefile.defines	2011-12-23 05:38:08.353710600 -0600
@@ -32,3 +32,7 @@ SHARED_INCLUDES = \
 
 INCLUDES += $(API_DEFINES)
 CXXFLAGS += $(API_DEFINES)
+
+ifeq ($(SHARED_GLAPI),1)
+DRI_LIB_DEPS += -L$(TOP)/$(LIB_DIR) -l$(GLAPI_LIB)
+endif
--- origsrc/Mesa-7.11.2/src/mesa/drivers/osmesa/Makefile	2011-07-08 20:31:41.000000000 -0500
+++ src/Mesa-7.11.2/src/mesa/drivers/osmesa/Makefile	2011-12-25 20:43:55.768839300 -0600
@@ -25,6 +25,10 @@ CORE_MESA = \
 	$(TOP)/src/mapi/glapi/libglapi.a \
 	$(TOP)/src/glsl/libglsl.a
 
+ifeq ($(SHARED_GLAPI),1)
+OSMESA_LIB_DEPS += -L$(TOP)/$(LIB_DIR) -l$(GLAPI_LIB)
+endif
+
 .c.o:
 	$(CC) -c $(INCLUDE_DIRS) $(CFLAGS) $< -o $@
 
