--- origsrc/Mesa-7.11.2/src/mesa/drivers/dri/common/xmlconfig.c	2011-10-14 19:43:58.000000000 -0500
+++ src/Mesa-7.11.2/src/mesa/drivers/dri/common/xmlconfig.c	2012-01-30 10:48:15.563427600 -0600
@@ -47,6 +47,8 @@
 extern char *program_invocation_name, *program_invocation_short_name;
 #    endif
 #    define GET_PROGRAM_NAME() program_invocation_short_name
+#elif defined(__CYGWIN__)
+#    define GET_PROGRAM_NAME() program_invocation_short_name
 #elif defined(__FreeBSD__) && (__FreeBSD__ >= 2)
 #    include <osreldate.h>
 #    if (__FreeBSD_version >= 440000)
